File: Sharebook/Server/src/Sharebook.Common/Public/Models/TweetIdentifier.cs
namespace Sharebook.Common.Public.Models
{
    using System.Globalization;

    using Sharebook.Common.Public.Models.Interfaces;

    public class TweetIdentifier : ITweetIdentifier
    {
        public TweetIdentifier(long tweetId)
        {
            Id = tweetId;
            IdStr = tweetId.ToString(CultureInfo.InvariantCulture);
        }

        public long Id { get; set; }
        public string IdStr { get; set; }

        public override string ToString()
        {
            return IdStr ?? Id.ToString();
        }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/TweetIdentifier.ts
import {Injectable} from "@angular/core";

import {ITweetIdentifier} from "./Interfaces/ITweetIdentifier";

@Injectable({
  providedIn: 'root',
})
export class TweetIdentifier implements ITweetIdentifier {
  constructor(tweetId: number) {
    this.id = tweetId;
    this.idStr = tweetId.toString(/*CultureInfo.InvariantCulture*/);
  }

  public id: number;
  public idStr: string;

  public toString(): string {
    return this.idStr ?? this.id.toString();
  }
}
